File: backend/src/models/kanban/CardCycleTime.ts
import { JsonObject, JsonProperty } from "json2typescript";

export class CardCycleTime {
  name: string;
  steps: StepsDay;
  total: number;

  constructor(name: string, steps: StepsDay, total: number) {
    this.name = name;
    this.steps = steps;
    this.total = total;
  }
}

@JsonObject("StepsDay")
export class StepsDay {
  @JsonProperty("analyse")
  analyse: number = 0;
  @JsonProperty("development")
  development: number = 0;
  @JsonProperty("waiting")
  waiting: number = 0;
  @JsonProperty("testing")
  testing: number = 0;
  @JsonProperty("blocked")
  blocked: number = 0;
  @JsonProperty("review")
  review: number = 0;
}

File: backend/src/main/java/heartbeat/controller/board/vo/response/CardCycleTime.java
package heartbeat.controller.board.vo.response;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class CardCycleTime {

	private String name;

	private StepsDay steps;

	private int total;

}

@Data
@Builder
class StepsDay {

	private int analyse;

	private int development;

	private int waiting;

	private int testing;

	private int blocked;

	private int review;

}
